### ISSUE TITLE
[KeePassXC] Allow more control over the presented information when searching for records

### ISSUE THREAD

**Issue by yaronuliel** [raycast/extensions#10709]
### Extension

https://www.raycast.com/zkytech/keepassxc

### Description

Currently, when searching for a password - the results show the title of the password, and the path to the records inside the directories as tags, as shown in the following image

<img width="752" alt="image" src="https://github.com/raycast/extensions/assets/5229872/486bef90-cfe0-4e9b-a170-9b75b09a1ac2">

Inside KeePassXC - a lot more information is shown on the search results, which help a lot when trying to understand what record to pick

I think it would be very helpfull if the extension allowes some configration of "what information to show" - including 

* Username
* URL
* Has TOTP?

As a  good start, showing the Username is crucial for picking the correct result.

Also, it might save some space if the path to the records is shown in a "breadcrumbs" format rather then individual tags (i.e. `root dir > dir1 > dir2 > dir 3`

### Who will benefit from this feature?

Every user of the extension, especially those that have big KeePass files, with multiple records with the same title, differing in other property, such as username

**Comment by Sulfyderz** [comment]
Hello,

Thank you for your feedback. I am a contributor to the extension, not the original creator (i.e. @zkytech).

> I think it would be very helpfull if the extension allows some configration of "what information to show" - including
> 
> Username
> URL
> Has TOTP?
> As a good start, showing the Username is crucial for picking the correct result.

Well, I do agree with you for the Username. Currently, I am using a workaround. My titles contain the username when there are homonyms (e.g. 'Google (Sulfyderz@mail.com)' and 'Google (Boba@mail.com)'). I think it would be a great feature to add that information.

Concerning the URL, I don't think it's that useful. Give me an example, maybe I'm wrong.

About the TOTP, it can be more or less useful. Why more or less? When you are on a website, you'll be prompted anyway to insert a TOTP code (you can then apply the code with `opt + enter` on the entry from the extension). Before that moment, you don't really need to know if an input does have a TOTP set. Still, give me an example in which case you find it insightful.

> Also, it might save some space if the path to the records is shown in a "breadcrumbs" format rather then individual tags (i.e. root dir > dir1 > dir2 > dir 3

I had repositories inside my keepassXC, before I use that extension. Not anymore. But yes, I do agree with you that your idea is a best UI.

If I have some time, I would gladly do all of this. But currently, I don't. May be in the end of 2024. @zkytech seems kind of busy as well. What I suggest is to do a PR.

Best,
Sulfyderz.

**Comment by yaronuliel** [comment]
@Sulfyderz  Referencing your comment:

>Well, I do agree with you for the Username. Currently, I am using a workaround. My titles contain the username when there are homonyms (e.g. 'Google ([Sulfyderz@mail.com](mailto:Sulfyderz@mail.com))' and 'Google ([Boba@mail.com](mailto:Boba@mail.com))'). I think it would be a great feature to add that information.

I have a KeePass database with thousands of records  - so this is practically impossible for me to do without scripting, and as you said, it is just a workaround


> About the TOTP, it can be more or less useful. Why more or less? When you are on a website, you'll be prompted anyway to insert a TOTP code (you can then apply the code with opt + enter on the entry from the extension). Before that moment, you don't really need to know if an input does have a TOTP set. Still, give me an example in which case you find it insightful.

when I use the extension to get TOTP (for example for an AWS account) - I find myself against 7 different records, and I need to guess which one contains the TOTP.
It is true that adding the username to the list would help with this concern as well, but still, I think that adding an emoji (🕔) to mark that a TOTP is set on the record is still beneficial.
I literally encounter this on a daily basis.

Regarding the implementation of the solution - I've looked into it and started working on it, and it seems that the KeePass CLI API, which is used to query the database, doesn't support returning an additional information using the search command, thus, in order to present the username/TOTP presence/URL - an additional command execution is required *for each record* in the list. The implementation is possible, but since it might add a lot of command executions, a more thorough investigation of the consequences on performance should be made before it is implemented

I did open an issue on the KeePassXC repo https://github.com/keepassxreboot/keepassxc/issues/10294 to add the option to return more info using the `search` command
